### PR TITLE
[bitnami/prestashop] New major version: bump mariadb and standarizations

### PR DESCRIPTION
**Pull request by marcosbc**
**Description of the change**

This PR bumps PrestaShop major version since its dependency (MariaDB) bumped the major version including breaking changes. More info at #3849

**Benefits**

- Standardization.
- Reduces technical debt.

**Possible drawbacks**

It breaks backwards compatibility.

**Applicable issues**

None

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Review by miguelaeh — APPROVED** [review]
Good job!!
LGTM!!

File: bitnami/prestashop/values.yaml [hunk 1]
```diff
@@ -477,6 +547,50 @@ metrics:
     prometheus.io/scrape: "true"
     prometheus.io/port: "9117"
 
+# Add custom certificates and certificate authorities to redmine container
```
Comment (javsalgar) [inline bitnami/prestashop/values.yaml:550]: redmine -> prestashop

File: bitnami/prestashop/values.yaml [hunk 2]
```diff
@@ -477,6 +547,50 @@ metrics:
...
+certificates:
+  customCertificate:
+    certificateSecret: ""
+    chainSecret: {}
+    # name: secret-name
+    # key: secret-key
+    certificateLocation: /etc/ssl/certs/ssl-cert-snakeoil.pem
+    keyLocation: /etc/ssl/private/ssl-cert-snakeoil.key
+    chainLocation: /etc/ssl/certs/mychain.pem
```
Comment (javsalgar) [inline bitnami/prestashop/values.yaml:559]: Could you these paths?

File: bitnami/prestashop/values.yaml (hunk 1 above)
Comment (marcosbc) [reply bitnami/prestashop/values.yaml]: Good catch, fixed

File: bitnami/prestashop/values.yaml (hunk 2 above)
Comment (marcosbc) [reply bitnami/prestashop/values.yaml:559]: This is using the defaults, basing on the work in #2774

> Adds a snakeoil certificate to the container. Used can be used for TLS sessions to the ingress.

File: bitnami/prestashop/templates/deployment.yaml [hunk 3]
```diff
@@ -51,6 +56,77 @@ spec:
...
+          - if command -v apk >/dev/null; then apk add --no-cache ca-certificates openssl && update-ca-certificates;
+            else apt-get update && apt-get install -y ca-certificates openssl; fi
+          {{- else }}
+          - sh
+          - -c
+          - if command -v apk >/dev/null; then apk add --no-cache ca-certificates openssl && update-ca-certificates;
+            else apt-get update && apt-get install -y ca-certificates openssl; fi
+            && openssl req -new -x509 -days 3650 -nodes -sha256
+              -subj "/CN=$(hostname)" -addext "subjectAltName = DNS:$(hostname)"
+              -out  /etc/ssl/certs/ssl-cert-snakeoil.pem
```
Comment (javsalgar) [inline bitnami/prestashop/templates/deployment.yaml:103]: Shouldn't this use the certificate location value?

**Review by javsalgar — APPROVED** [review]
Minor thing, LGTM

File: bitnami/prestashop/templates/deployment.yaml (hunk 3 above)
Comment (marcosbc) [reply bitnami/prestashop/templates/deployment.yaml:103]: good point, changing

File: bitnami/prestashop/templates/deployment.yaml [hunk 4]
```diff
@@ -85,36 +161,22 @@ spec:
...
             - name: PRESTASHOP_DATABASE_USER
-              value: {{ .Values.mariadb.db.user | quote }}
+              value: {{ include "prestashop.databaseUser" . | quote }}
             - name: PRESTASHOP_DATABASE_PASSWORD
               valueFrom:
                 secretKeyRef:
-                  name: {{ template "prestashop.mariadb.fullname" . }}
+                  name: {{ include "prestashop.databaseSecretName" . }}
                   key: mariadb-password
-            {{- else }}
```
Comment (lupieczonka) [inline bitnami/prestashop/templates/deployment.yaml:102]: externaldb-secrets.yaml doesn't work now

Comment (dani8art) [reply bitnami/prestashop/templates/deployment.yaml:102]: Hi could you open an issue and explain it there?

Comment (lupieczonka) [reply bitnami/prestashop/templates/deployment.yaml:102]: Hi, the external database is using a different password key
externaldb-secrets.yaml create: secrets
name: "common.names.fullname-externaldb" 
key: db-password

path:
            {{- if .Values.mariadb.enabled }}
            - name: PRESTASHOP_DATABASE_HOST
              value: {{ include "prestashop.databaseHost" . | quote }}
            - name: PRESTASHOP_DATABASE_PORT_NUMBER
              value: {{ include "prestashop.databasePort" . | quote }}
            - name: PRESTASHOP_DATABASE_NAME
              value: {{ include "prestashop.databaseName" . | quote }}
            - name: PRESTASHOP_DATABASE_USER
              value: {{ include "prestashop.databaseUser" . | quote }}
            - name: PRESTASHOP_DATABASE_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: {{ include "prestashop.databaseSecretName" . }}
                  key: mariadb-password
            {{- else }}
            - name: PRESTASHOP_DATABASE_HOST
              value: {{ include "prestashop.databaseHost" . | quote }}
            - name: PRESTASHOP_DATABASE_PORT_NUMBER
              value: {{ include "prestashop.databasePort" . | quote }}
            - name: PRESTASHOP_DATABASE_NAME
              value: {{ include "prestashop.databaseName" . | quote }}
            - name: PRESTASHOP_DATABASE_USER
              value: {{ include "prestashop.databaseUser" . | quote }}
            - name: PRESTASHOP_DATABASE_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: {{ printf "%s-%s" .Release.Name "externaldb" }}
                  key: db-password
            {{- end }}

Comment (dani8art) [reply bitnami/prestashop/templates/deployment.yaml:102]: I've just create a PR https://github.com/bitnami/charts/pull/4026, please, let's move the conversation there. Thanks

**Pull request merged**